### PR TITLE
dotnet-validate: Add initial support for glob patterns in local package validation...

### PR DESCRIPTION
**Pull request by danielcweber**
... with the help of Microsoft.Extensions.FileSystemGlobbing. Can handle fully qualified and relative paths, but there might be edge cases.

### CODE REVIEW

File: dotnet-validate/Program.cs [hunk 1]
```diff
@@ -52,35 +53,56 @@ private static async Task<int> Main(string[] args)
...
-            }
-            catch (Exception exception)
+            foreach (var actualFile in files.Files)
             {
-                await Console.Error.WriteLineAsync(exception.ToString()).ConfigureAwait(false);
-                return EX_SOFTWARE;
+                try
+                {
+                    using var cancellationTokenSource = new CancellationTokenSource();
+                    Console.CancelKeyPress += (_, eventArgs) =>
```
Comment (campersau) [inline dotnet-validate/Program.cs:76]: Can this event handler and the `cancellationTokenSource` be moved outside of the `foreach`? And then maybe add a `cancellationTokenSource.Token.IsCancellationRequested` check inside of the loop.

**Comment by clairernovotny** [comment]
Probably! This CLI needs a lot of love, I was just trying to get something working out :)

File: dotnet-validate/Program.cs (hunk 1 above)
Comment (danielcweber) [reply dotnet-validate/Program.cs]: Good point. Was completely oblivious about that cancellationtoken. Done.

**Comment by danielcweber** [comment]
@clairernovotny Much appreciated! I was about to include it into my pipeline for [Gremlinq](https://github.com/ExRam/ExRam.Gremlinq) this morning, but it produces about 12 packages so this was a welcome opportunity to contribute.

**Comment by clairernovotny** [comment]
Is this ready to merge?

**Comment by danielcweber** [comment]
@clairernovotny As far as I am concerned, yes.

**Comment by danielcweber** [comment]
@clairernovotny Would you mind a new preview release? I'd include it in my github actions build.

**Comment by clairernovotny** [comment]
Done

**Comment by 0xced** [comment]
This pull request makes it impossible to validate a local package with an absolute path. 😕

I think I fixed the issue by replacing `var packageFile = new FileInfo(actualFile.Path);` with `var packageFile = new FileInfo(Path.Combine(directory, actualFile.Path));` but I'm not familiar with `Microsoft.Extensions.FileSystemGlobbing` so I'm not really sure if that's the correct way to fix this issue.

**Comment by danielcweber** [comment]
@0xced This should still be possible through [this if-branch](https://github.com/NuGetPackageExplorer/NuGetPackageExplorer/pull/1213/files#diff-a3d25c9db4327ab20adfe0267ffc0630ad9e6a234ea8a1d57510762655fb8cecR58) which worked well for me locally (Windows) but as I said, there might be edge cases. Could you post your solution and an absolute path that's broken for you?

**Comment by campersau** [comment]
@danielcweber  I can observe the same issue as @0xced 
I used the `package local` command with the following files:

```
\\server\folder\nuget\test.nupkg

directory: \\server\folder
file: nuget\test.nupkg

actualFile.Path: nuget/test.nupkg
```

```
D:\test.nupkg

directory: D:\
file: test.nupkg

actualFile.Path: test.nupkg
```

```
D:\*.nupkg

directory: D:\
file: *.nupkg

actualFile.Path: test.nupkg
```

As you can see the `actualFile.Path` only contains the filename in these cases, not the full path so a `Path.Combine(directory, actualFile.Path)` would fix that.

**Comment by danielcweber** [comment]
@campersau @0xced #1229 for review

**Pull request merged**